### PR TITLE
examples: zephyr: hello_nrf91_offloaded: new example

### PR DESCRIPTION
**Pull request by mniestroj**
Add minimal example that shows how to use nRF91 with DTLS offloaded
sockets.

### CODE REVIEW

File: examples/zephyr/hello_nrf91_offloaded/prj.conf
Comment (sam-golioth) [inline examples/zephyr/hello_nrf91_offloaded/prj.conf:1]: Do we need to set `CONFIG_GOLIOTH_SAMPLE_AUTH_TYPE_TAG`?

Comment (mniestroj) [reply examples/zephyr/hello_nrf91_offloaded/prj.conf:1]: You are right. Both work, but setting `CONFIG_GOLIOTH_SAMPLE_AUTH_TYPE_TAG=y` is definitely better. It saves some runtime manipulation of the tls_credential Zephyr subsystem.

File: examples/zephyr/hello_nrf91_offloaded/Kconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021 Golioth, Inc.
```
Comment (sam-golioth) [inline examples/zephyr/hello_nrf91_offloaded/Kconfig:1]: 2024

File: examples/zephyr/hello_nrf91_offloaded/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+[00:00:00.070,000] <inf> golioth_coap_client: Session PSK-ID: your-device-id@your-golioth-project
+[00:00:00.070,000] <inf> golioth_coap_client: Entering CoAP I/O loop
+[00:00:01.260,000] <inf> golioth_coap_client: Golioth CoAP client connected
+[00:00:01.260,000] <inf> hello_zephyr: Sending hello! 0
+[00:00:01.260,000] <inf> hello_zephyr: Golioth client connected
+[00:00:06.270,000] <inf> hello_zephyr: Sending hello! 1
+[00:00:11.280,000] <inf> hello_zephyr: Sending hello! 2
+```
+
+Responses to Hello messages are printed above as a hexdump of \"Hello
```
Comment (sam-golioth) [inline examples/zephyr/hello_nrf91_offloaded/README.md:107]: Is this line true?

**Review by sam-golioth — COMMENTED** [review]
The sample test failed in CI. Looks like an issue invoking pytest, but I didn't look deeply.

**Review by sam-golioth — APPROVED** [review]

**Pull request merged**